src/mapping/config/MappingConfiguration.cpp

```diff
@@ -538,6 +553,23 @@ MappingConfiguration::ConfiguredMapping MappingConfiguration::createMapping(
   ConfiguredMapping configuredMapping;
   mesh::PtrMesh     fromMesh(_meshConfig->getMesh(fromMeshName));
   mesh::PtrMesh     toMesh(_meshConfig->getMesh(toMeshName));
+
+  // Handle for just-in-time mapping, we copy over the dimension and leave everything else
+  if (fromMesh.get() == nullptr && fromMeshName.empty()) {
+    PRECICE_CHECK(toMesh.get() != nullptr,
+                  "Mesh \"{0}\" was not found while creating a mapping. "
+                  "Please correct the to=\"{0}\" attribute.",
+                  toMeshName);
+    fromMesh = mesh::MeshConfiguration::getJustInTimeMappingMesh(toMesh->getDimensions());
+  }
+  if (toMesh.get() == nullptr && toMeshName.empty()) {
+    PRECICE_CHECK(fromMesh.get() != nullptr,
```

Comment:
This should be equivalent

```suggestion
  if (!fromMesh && fromMeshName.empty()) {
    PRECICE_CHECK(toMesh,
                  "Mesh \"{0}\" was not found while creating a mapping. "
                  "Please correct the to=\"{0}\" attribute.",
                  toMeshName);
    fromMesh = mesh::MeshConfiguration::getJustInTimeMappingMesh(toMesh->getDimensions());
  }
  if (!toMesh && toMeshName.empty()) {
    PRECICE_CHECK(fromMesh,
```

Reply:
Applied it locally to also address the other spots